mancala/server/src/lib.rs

```diff
@@ -1,24 +1,36 @@
 use std::io::{Read, Write};
 use std::net::{Shutdown, TcpListener, TcpStream};
+use std::str;
 use std::thread;
 
 fn handle_client(mut stream: TcpStream) {
-    let mut data = [0 as u8; 50]; // using 50 byte buffer
-    while match stream.read(&mut data) {
-        Ok(size) => {
-            // echo everything!
-            stream.write_all(&data[0..size]).unwrap();
-            true
-        }
-        Err(_) => {
-            println!(
-                "An error occurred, terminating connection with {}",
-                stream.peer_addr().unwrap()
-            );
-            stream.shutdown(Shutdown::Both).unwrap();
-            false
+    let mut buffer = [0; 512]; // use a 512 byte buffer
+    loop {
+        match stream.read(&mut buffer) {
+            Ok(size) => {
+                let input = str::from_utf8(&buffer[0..size]).unwrap().trim_end();
+                println!("Data received: {}", input);
+                // Exit loop & terminate connection if user enters "quit"
+                if input.eq_ignore_ascii_case("quit") {
```

Comment:
I think we need to check for a connection unexpectedly terminated here as well.  If you start up the client / server, then exit the client with ctrl+c rather than "quit", which is a reasonable scenario, the server doesn't terminate the connection and just loops on "Data recieved: " forever

Reply:
good point. I hadn't tested the lost connection scenario